backend/matching-service/src/index.js

```diff
@@ -0,0 +1,116 @@
+import express from 'express';
+import dotenv from 'dotenv';
+import cors from 'cors';
+import { connectMongoDB, connectRabbitMQ, getChannel } from './connections.js';
+import { addUser, getMatchQueue, removeUser, initializeMatchQueue } from './matchQueue.js';
+import { initializeWebSocketServer, notifyUser } from './websocket.js'
+
+dotenv.config();
+
+const app = express();
+let channel;
+
+app.use(express.json());
+app.use(cors()); 
+
+const initialize = async () => {
+  await connectMongoDB();
+  await connectRabbitMQ();
+  channel = getChannel();
+  await initializeMatchQueue();
+  await processMatchRequests();
+};
+
+initialize().catch(console.error);
+
+initializeWebSocketServer(process.env.WEBSOCKET_PORT || 8080)
+
+const processMatchRequests = async () => {
+  await channel.consume('match_requests', async (request) => {
+    if (request !== null) {
+      const matchRequest = JSON.parse(request.content.toString());
+      const matchQueue = await getMatchQueue();
+      let matched = false;
+
+      // Matches with the same topic and difficulty
+      for (let i = 0; i < matchQueue.length; i++) {
+        if (
+          matchQueue[i].topic === matchRequest.topic &&
+          matchQueue[i].difficulty === matchRequest.difficulty &&
+          matchQueue[i].userId !== matchRequest.userId
```

Comment:
There are three conditions within this if-block. You may want to consider creating a boolean function that checks the matching criteria for better readability.

As an extension, you may want to consider having two types of such boolean functions:
1. `meetHardMatchingCriteria()`: Both `topic` and `difficulty` of the two users must equal before they are matched.
2. `meetSoftMatchingCriteria()`: Only `topic` of the two users are equal.